### PR TITLE
Keep request counts low

### PR DESCRIPTION
**Pull request by markus-ntt-seidl**
Conversion of #58

I converted them to two patterns, as the second one (small transfer sizes) is a combination of all minification patterns/deprecate-gifs/use modern image formats/etc.

### CODE REVIEW

**Comment by dubrie** [comment]
Looks good from initial review

**Comment by dubrie** [comment]
assigning to @greenhsu123 for SME review on this

File: docs/catalog/web/keep-transfer-sizes-small.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+`SCI = (E * I) + M per R`  
+[Software Carbon Intensity Spec](https://grnsft.org/sci)
+
+Concerning the SCI equation, keeping the request counts low will impact:
+
+- `I`: Reducing the file size of resources also reduces bandwidth and, maybe, parsing and processing time will also be reduced
+- `M`: Reducing the file size on each resource reduces the storage needed
+
+
+## Assumptions
```
Comment (greenhsu123) [inline docs/catalog/web/keep-transfer-sizes-small.md:40]: Assumptions?

File: docs/catalog/web/keep-transfer-sizes-small.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+
+## Considerations
```
Comment (greenhsu123) [inline docs/catalog/web/keep-transfer-sizes-small.md:43]: Considerations?

File: docs/catalog/web/keep-transfer-sizes-small.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+# Keep transfer sizes small
+
+## Description
+
+Displaying a web page requires that the browser has downloaded all necessary files needed. Not only does the amount of request determine how much CO2 is used, but also the file sizes of each request.
+
+To use the least amount of CO2 it is more efficient to download a larger, completely necessary, resource instead of multiple tiny resources. 
+
+## Solution
```
Comment (greenhsu123) [inline docs/catalog/web/keep-transfer-sizes-small.md:20]: I think it's best if we split those out. Would you mind also explaining what you mean by (TODO Link to pattern)?

File: docs/catalog/web/keep-transfer-sizes-small.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+## Assumptions
+
+
+## Considerations
+
+
+## References
+
+- [Source](https://web.dev/resource-summary/)
```
Comment (greenhsu123) [inline docs/catalog/web/keep-transfer-sizes-small.md:48]: Could we link the reference to a specific subheading of the page? And be more descriptive than just "source"?

File: docs/catalog/web/keep-request-counts-low.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+## Assumptions
+
+
+## Considerations
+
+
+## References
+
+- [Source](https://web.dev/resource-summary/)
```
Comment (greenhsu123) [inline docs/catalog/web/keep-request-counts-low.md:42]: Could we link the reference to a specific subheading of the page? And be more descriptive than just "source"?

**Comment by greenhsu123** [comment]
Agree in the meeting to split the PR into two. 
Meta-pattern discussion is to b continued.

File: docs/catalog/web/keep-transfer-sizes-small.md (hunk 4 above)
Comment (markus-ntt-seidl) [reply docs/catalog/web/keep-transfer-sizes-small.md:48]: There is not a more descriptive section in the source to this pattern.

File: docs/catalog/web/keep-request-counts-low.md (hunk 5 above)
Comment (markus-ntt-seidl) [reply docs/catalog/web/keep-request-counts-low.md:42]: There is not a more descriptive section in the source to this pattern.

File: docs/catalog/web/keep-request-counts-low.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+Reduce the amount of files needed and also the amount of requests done by the loaded JavaScript to display the final page. 
+
+## SCI Impact
+
+`SCI = (E * I) + M per R`  
+[Software Carbon Intensity Spec](https://grnsft.org/sci)
+
+Concerning the SCI equation, keeping the request counts low will impact:
+
+- `I`: Reducing the amount of requests reduces the energy needed to display the page
```
Comment (greenhsu123) [inline docs/catalog/web/keep-request-counts-low.md:30]: Isn't I is carbon intensity?

File: docs/catalog/web/keep-request-counts-low.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+## Assumptions
+
+
+## Considerations
+
+
+## References
+
+- [Detailed description](https://web.dev/resource-summary/)
```
Comment (greenhsu123) [inline docs/catalog/web/keep-request-counts-low.md:42]: Could we include what topics are referenced in this pattern? i.e. picked the topics specifically apply to this pattern from https://web.dev/resource-summary/?

File: docs/catalog/web/keep-request-counts-low.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+Concerning the SCI equation, keeping the request counts low will impact:
+
+- `I`: Reducing the amount of requests reduces the energy needed to display the page
+- `M`: Reducing the amount of requests can also reduce the amount of files and therefore the storage requirements of the web page on the server
+
+
+## Assumptions
+
+
+## Considerations
```
[the remark this answers is not in the document]
Comment (greenhsu123) [reply docs/catalog/web/keep-request-counts-low.md:37]: How about encouraging readers to consider [setting performance budgets](https://web.dev/fast#set-performance-budgets) so any future regression can be prevented?

File: docs/catalog/web/keep-request-counts-low.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+`SCI = (E * I) + M per R`  
+[Software Carbon Intensity Spec](https://grnsft.org/sci)
+
+Concerning the SCI equation, keeping the request counts low will impact:
+
+- `I`: Reducing the amount of requests reduces the energy needed to display the page
+- `M`: Reducing the amount of requests can also reduce the amount of files and therefore the storage requirements of the web page on the server
+
+
+## Assumptions
```
[the remark this answers is not in the document]
Comment (greenhsu123) [reply docs/catalog/web/keep-request-counts-low.md:34]: Are we sort of assuming that the server hosting those files is not inherently slow?

[the remark this answers is not in the document]
Comment (markus-ntt-seidl) [reply docs/catalog/web/keep-request-counts-low.md:34]: "Are we sort of assuming that the server hosting those files is not inherently slow? " No, not directly. We are assuming that a lot of files transferred raises the SCI. Performance is a side effect from SCI point of view. The sources of all the web patterns (web.dev) I saw here, has the focus on performance optimization. Which is related to SCI, but not directly applicable in all cases.
To be clear, I don't think this pattern involves any optimization on the server part regularly (except if you dynamically generate javascript/css/etc on the server)

File: docs/catalog/web/keep-request-counts-low.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (markus-ntt-seidl) [reply docs/catalog/web/keep-request-counts-low.md:37]: Good point! Will include it.

File: docs/catalog/web/keep-request-counts-low.md (hunk 7 above)
Comment (markus-ntt-seidl) [reply docs/catalog/web/keep-request-counts-low.md:42]: We focus on request count but the source does not have this, so there is no part to directly link to that makes this more specific. The chapters in the source are for each resource type, linking to one resource type directly doesn't benefit the pattern.

**Comment by dubrie** [comment]
This looks to have addressed the concerns for the pattern, @greenhsu123 is it ready for SME approval?

**Comment by greenhsu123** [comment]
LGTM- completing SME review.

**Comment by dubrie** [comment]
Moved to Technical Writer Review state, awaiting feedback from the technical writer

**Comment by holanita** [comment]
Waiting for further PRs. We'll to the technical review in one batch at the end of the year or early next year.

**Pull request merged**